<?xml version="1.0"?>
<launch>
    <!-- the root of the name for each camera topic which will be followed by the device number -->
    <arg name="camera_name" default="camera" />
    <!-- video_stream_provider has to be a number or an array of numbers of video devices -->
    <!-- for example 0 or [0, 2, 4, 6, 8] -->
    <arg name="video_stream_provider" default="0" />
    <!-- set buffer queue size of frame capturing to -->
    <arg name="buffer_queue_size" default="10" />
    <!-- frames per second for publish -->
    <arg name="fps" default="10" />
    <!-- frame_id for the camera -->
    <arg name="frame_id" default="world" />
    <!-- defines if the script should try to automatically reconnect if the frame was not captured -->
    <arg name="retry_on_fail" default="false" />

    <node pkg="video_stream_opencv" type="node.py" name="$(arg camera_name)_stream" output="screen">
        <param name="camera_name" type="string" value="$(arg camera_name)" />
        <param name="video_stream_provider" type="string" value="$(arg video_stream_provider)" />
        <param name="buffer_queue_size" type="int" value="$(arg buffer_queue_size)" />
        <param name="fps" type="double" value="$(arg fps)" />
        <param name="frame_id" type="string" value="$(arg frame_id)" />
        <param name="retry_on_fail" type="bool" value="$(arg retry_on_fail)" />
    </node>

</launch>
